driver-core/src/test/unit/com/mongodb/internal/connection/InternalStreamConnectionInitializerSpecification.groovy

```diff
@@ -243,6 +256,94 @@ class InternalStreamConnectionInitializerSpecification extends Specification {
                                  [true, false]].combinations()
     }
 
+    @IgnoreIf({ !serverVersionAtLeast(4, 3) })
+    def 'should speculatively authenticate with default authenticator'() {
+        given:
+        def credential = new MongoCredentialWithCache(createCredential('user', 'database', 'pencil' as char[]))
+        def authenticator = Spy(DefaultAuthenticator, constructorArgs: [credential])
+        def scramShaAuthenticator = new ScramShaAuthenticator(credential.withMechanism(AuthenticationMechanism.SCRAM_SHA_256),
+                { 'rOprNGfwEbeRWgbNEkqO' }, { 'pencil' })
+        def initializer = new InternalStreamConnectionInitializer(authenticator, null, [])
+        authenticator.getAuthenticatorForIsMaster() >> scramShaAuthenticator
+
+        when:
+        enqueueSpeculativeAuthenticationResponsesForScramSha256()
+        def description = initializer.initialize(internalConnectionForSpeculativeAuthentication)
+
+        then:
```

Comment:
This should also assert that the `isMaster` command executed by the initializer contains the expected `speculativeAuthenticate` field.